Framework/DataHandling/inc/MantidDataHandling/MultiPeriodLoadMuonStrategy.h

```diff
@@ -0,0 +1,38 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+#include "MantidAPI/WorkspaceGroup_fwd.h"
+#include "MantidDataHandling/LoadMuonStrategy.h"
+#include "MantidNexus/NexusClasses.h"
+
+namespace Mantid {
+namespace DataHandling {
+class MultiPeriodLoadMuonStrategy : public LoadMuonStrategy {
+public:
+  // Constructor
+  MultiPeriodLoadMuonStrategy(Kernel::Logger &g_log, const std::string filename,
```

Comment:
`const std::string &filename`

Reply:
Hmm, so in a previous review of this Loader it was suggested by Martyn that I pass the filename by value rather than passing a reference to the local variable. So now a bit torn about which one to follow.

Reply:
Good point, Martyn is right on this as I'm used to pre-C++-17
Because of copy elision tooling (such as Clang-tidy) [now looks for this](https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html)